src/components/DeeplinkWrapper/index.website.js

```diff
@@ -0,0 +1,158 @@
+import CONST from '../../CONST';
+import CONFIG from '../../CONFIG';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import colors from '../../styles/colors';
+import * as Browser from '../../libs/Browser';
+
+const propTypes = {
+    /** Children to render. */
+    children: PropTypes.node.isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+class DeeplinkWrapper extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            appInstallationCheckStatus: this.isMacOSWeb() ? CONST.DESKTOP_DEEPLINK_APP_STATE.CHECKING : CONST.DESKTOP_DEEPLINK_APP_STATE.NOT_INSTALLED,
+        };
+    }
+
+    componentDidMount() {
+        if (!this.isMacOSWeb()) {
+            return;
+        }
+
+        let focused = true;
+
+        window.addEventListener('blur', () => {
+            focused = false;
+        });
+
+        setTimeout(() => {
+            if (!focused) {
+                this.setState({appInstallationCheckStatus: CONST.DESKTOP_DEEPLINK_APP_STATE.INSTALLED});
+            } else {
+                this.setState({appInstallationCheckStatus: CONST.DESKTOP_DEEPLINK_APP_STATE.NOT_INSTALLED});
+            }
+        }, 500);
+
+        // check if pathname matches with deeplink routes
+        const matchedRoute = _.find(deeplinkRoutes, (route) => {
+            const routeRegex = new RegExp(route.pattern);
+            return routeRegex.test(window.location.pathname);
+        });
+
+        if (matchedRoute) {
+            this.setState({deeplinkMatch: true});
+            this.openRouteInDesktopApp();
+        } else {
+            this.setState({deeplinkMatch: false});
+        }
+    }
+
+    openRouteInDesktopApp() {
+        const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
+        const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}${window.location.pathname}`;
+
+        // This check is necessary for Safari, otherwise, if the user
+        // does NOT have the Expensify desktop app installed, it's gonna
+        // show an error in the page saying that the address is invalid
+        if (CONST.BROWSER.SAFARI === Browser.getBrowser()) {
+            const iframe = document.createElement('iframe');
+            iframe.style.display = 'none';
+            document.body.appendChild(iframe);
+            iframe.contentWindow.location.href = expensifyDeeplinkUrl;
+
+            // Since we're creating an iframe for Safari to handle
+            // deeplink we need to give this iframe some time for
+            // it to do what it needs to do. After that we can just
+            // remove the iframe.
+            setTimeout(() => {
+                if (!iframe.parentNode) {
+                    return;
+                }
+
+                iframe.parentNode.removeChild(iframe);
+            }, 100);
+        } else {
+            window.location.href = expensifyDeeplinkUrl;
+        }
+    }
+
+    isMacOSWeb() {
```

Comment:
Why did we put this in the component instead of the `Browser` lib?

Reply:
No specific reasons we just missed out on that Review.